### PR TITLE
feat(route): add wechat/data258

### PR DESCRIPTION
**Pull request by Rongronggg9**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/wechat/data258
/wechat/data258/caifu
/wechat/data258/gh_cbbad4c1d33c
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [x] 反爬/频率限制 anti-bot or rate limit?
  - [x] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: docs/new-media.md [hunk 1]
```diff
@@ -3382,6 +3382,19 @@ column 为 third 时可选的 category:
...
 <Route author="sanmmm" example="/wechat/ershicimi/813oxJOl" path="/wechat/ershicimi/:id" :paramsDesc="['公众号id, 打开公众号页, 在 URL 中找到 id']" anticrawler="1"/>
 
+### 公众号 (微阅读来源)
+
+<Route author="Rongronggg9" example="/wechat/data258/gh_cbbad4c1d33c" path="/data258/:id?" :paramsDesc="['公众号 id 或分类 id，可在公众号页或分类页 URL 中找到；若略去，则抓取首页']" anticrawler="1" radar="1" rssbud="1" selfhost="1">
+
+::: warning 注意
+
+由于使用了一些针对反爬的缓解措施，本路由响应较慢。默认只抓取前 5 条，可通过 `?limit=` 改变（不推荐，容易被反爬）。\
+该网站使用 IP 甄别访客，且应用严格的每日阅读量限额 (不少于 15 次)，请自建并确保正确配置缓存；如使用内存缓存而非 Redis 缓存，请增大缓存容量。该限额足够订阅至少 3 个公众号 (假设公众号每日仅更新一次)；首页 / 分类页更新相当频繁，不推荐订阅。
```
Comment (TonyRL) [inline docs/new-media.md:3392]:
```suggestion
该网站使用 IP 甄别访客，且应用严格的每日阅读量限额 (不多于 15 次)，请自建并确保正确配置缓存；如使用内存缓存而非 Redis 缓存，请增大缓存容量。该限额足够订阅至少 3 个公众号 (假设公众号每日仅更新一次)；首页 / 分类页更新相当频繁，不推荐订阅。
```
typo?

File: lib/v2/wechat/data258.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+    // !!! double-check !!!
+    if ((await ctx.cache.get('data258:lock', false)) === '1') {
+        throw new RequestInProgressError('Another request is in progress, please try again later.');
+    } else {
+        // !!! here we acquire the lock because the jump page has crawler detection !!!
+        await ctx.cache.set('data258:lock', '1', 60);
+    }
+
+    // !!! here we must use a for-loop to ensure the concurrency is 1 !!!
```
Comment (TonyRL) [inline lib/v2/wechat/data258.js:69]: You can use `tiny-async-pool`

File: lib/v2/wechat/data258.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    const link = hyperlink.attr('href');
+    const pubDate = timezone(parseDate($item.find(timeSelector).text(), 'YYYY-MM-DD HH:mm'), 8);
+    return {
+        title,
+        link,
+        pubDate,
+    };
+};
+
+const sleep = (time) => new Promise((resolve) => setTimeout(resolve, time * 1000));
```
Comment (TonyRL) [inline lib/v2/wechat/data258.js:20]: `@/utils/wait` has the same functionality

File: lib/v2/wechat/data258.js (hunk 2 above)
Comment (Rongronggg9) [reply lib/v2/wechat/data258.js:69]: I consider there is no such need to use it if the concurrency is 1. Would you please describe the advantages of adopting it in this case? Making the flow body a function to improve the performance?

Comment (TonyRL) [reply lib/v2/wechat/data258.js:68]: No benefits. I was thinking this helps to tweak the concurrency(to 2) easily but after getting throttled multiple times I would consider a no for `tiny-async-pool`

**Pull request merged**